lib/iris/util.py

```diff
@@ -1080,6 +1080,10 @@ def new_axis(src_cube, scalar_coord=None):
     if scalar_coord is not None:
         scalar_coord = src_cube.coord(scalar_coord)
 
+    if src_cube.shape is ():
```

Comment:
My suggestion is to alter the actual slicing if we have a `MaskedConstant` rather than trying to correct for it beforehand. I tentatively suggest something like:

```python
if src_cube.has_lazy_data():
    new_cube = iris.cube.Cube(src_cube.lazy_data()[None])
else:
    if isinstance(src_cube.data, ma.core.MaskedConstant):
        new_data = ma.array([0], mask=[True])
    else:
        new_data = src_cube.data[None]
    new_cube = iris.cube.Cube(new_data)
```

with the following caveat: Do we know if this does the right thing when we have lazy data? I haven't checked. I think whatever implementation we choose we should have a test that is the equivalent of your new test but for a lazy array.